### PR TITLE
Test tl/2 and document use with improper lists

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -833,6 +837,17 @@ defmodule Kernel do
       tl([])
       #=> ** (ArgumentError) argument error
 
+      tl([:one])
+      #=> []
+
+  Given an improper list, it returns an improper list or the its last element.
```
Comment (whatyouhide) [inline lib/elixir/lib/kernel.ex]: Additional "the" here.

File: lib/elixir/lib/kernel.ex [hunk 2]
```diff
@@ -823,6 +823,10 @@ defmodule Kernel do
   @doc """
   Returns the tail of a list. Raises `ArgumentError` if the list is empty.
 
+  If a one-element list is passed, it returns an empty list.
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex]: I don't think we need this clarification. The tail of a one-element list is an empty list.

File: lib/elixir/lib/kernel.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/kernel.ex]: I don't think we need this clarification. The tail of an improper list is an improper list or the last element. The examples are welcome but those are properties and when listed in the documentation as so they feel like special cases.

**Comment by eksperimental** [comment]
all issues addressed

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/elixir/test/elixir/kernel_test.exs [hunk 3]
```diff
@@ -856,11 +856,25 @@ defmodule KernelTest do
     end
   end
 
+  test "tl/2" do
+    assert tl([:one]) == []
+    assert tl([1, 2, 3]) == [2, 3]
+
+    assert_raise ArgumentError, "argument error", fn ->
+      tl([])
```
Comment (eksperimental) [inline lib/elixir/test/elixir/kernel_test.exs]: Sorry folks, but this test is giving a warning,
> warning: this expression will fail with ArgumentError

Any idea how to bypass the warning?

Comment (eksperimental) [inline lib/elixir/test/elixir/kernel_test.exs]: solved: https://github.com/elixir-lang/elixir/pull/6006

**Pull request merged**